app/scripts/lib/rpc-method-middleware/handlers/watch-asset.js

```diff
@@ -36,9 +37,11 @@ async function watchAssetHandler(
 ) {
   try {
     const { options: asset, type } = req.params;
-    res.result = await handleWatchAssetRequest(asset, type);
+    const handleWatchAssetResult = await handleWatchAssetRequest(asset, type);
+    const result = await handleWatchAssetResult.result;
+    res.result = Boolean(result);
     return end();
   } catch (error) {
-    return end(error);
+    return end(ethErrors.provider.userRejectedRequest());
```

Comment:
We can look for this error specifically. Right now the controller returns an error with the message "User rejected to watch the asset." if the user rejects.

Right now this is effectively swallowing the `error`, hiding any potential unexpected error that might be thrown here. Better to bubble it up as an internal error if we don't recognize it.